orttraining/tools/onnx2tfevents/onnx2tfevents.py

```diff
@@ -0,0 +1,379 @@
+                        if "/" not in output:
+                            self.new_names[output] = prefix + output
+
+
+class YieldOpTransformer(ReplaceNameTransformer):
+    """Handle the output names without hierarchical information that are introduced by YieldOp."""
+
+    def _generate_new_names(self, graph):
+        for node in graph.node:
+            if node.op_type == "YieldOp":
+                # TODO: the length of input and output can be not equal if attribute non_differentiable_outputs
+                # is not empty. Need to handle this case.
+                assert len(node.input) == len(node.output)
+                for idx in range(len(node.input)):
+                    prefix = get_prefix(node.input[idx])
+                    if len(prefix) > 0:
+                        self.new_names[node.output[idx]] = prefix + node.output[idx]
+
+
+class ListUnpackTransformer(TransformerBase):
+    """Tensorboard's NodeDef doesn't have output, it assumes each node has only one output and put this output as
+    node name. If an ONNX node have more than one outputs, we will add a special node named ListUnpack for each output.
+    """
+
+    def __init__(self):
+        self.ops = {}
+
+    def transform(self, graph):
+        new_nodes = []
+        for node in graph.node:
+            if len([output for output in node.output if len(output) > 0]) > 1:
+                idx = 0
+                if node.op_type in self.ops:
+                    idx = self.ops[node.op_type]
+                    self.ops[node.op_type] = idx + 1
+                else:
+                    self.ops[node.op_type] = 1
+                new_output = get_prefix(node.output[0]) + node.op_type + "_" + str(idx) + "_output"
+                for output in node.output:
+                    if len(output) > 0:
+                        new_nodes.append(helper.make_node("ListUnpack", [new_output], [output]))
+                node.ClearField("output")
+                node.output.extend([new_output])
+        if len(new_nodes) > 0:
+            graph.node.extend(new_nodes)
+
+
+class AppendOpTypeTransformer(ReplaceNameTransformer):
+    """Tensorboard's node search can only index the node name, not the node type. To make the op type searchable,
+    append the op type to the end of the node name.
+    """
+
+    def _generate_new_names(self, graph):
+        for node in graph.node:
+            self.new_names[node.output[0]] = node.output[0] + "::" + node.op_type
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--logdir", help="Tensorboard log directory")
+    parser.add_argument("--model", help="ONNX model path")
+
+    args = parser.parse_args()
+    if not args.logdir or not args.model:
+        print("Fail to convert. Please specify the tensorboard log directory and the onnx model path.")
+        print("Usage: python3 onnx2tfevents.py --logdir <tensorboard log directory> --model <onnx model path>")
```

Comment:
Argparser will generate this message. Was there a reason it needs to be constructed here?